### PR TITLE
JIT: Fix ShiftRightLogical constant folding

### LINKED ISSUE #105558: JIT: Another potential bad codegen with shift by `Vector128`
```csharp
// Generated by Fuzzlyn v2.1 on 2024-07-26 12:56:41
// Run on X64 Windows
// Seed: 13544108888657591911-vectort,vector128,vector256,x86aes,x86avx,x86avx2,x86bmi1,x86bmi1x64,x86bmi2,x86bmi2x64,x86fma,x86lzcnt,x86lzcntx64,x86pclmulqdq,x86popcnt,x86popcntx64,x86sse,x86ssex64,x86sse2,x86sse2x64,x86sse3,x86sse41,x86sse41x64,x86sse42,x86sse42x64,x86ssse3,x86x86base
// Reduced from 290.9 KiB to 0.6 KiB in 00:02:37
// Debug: Prints 1 line(s)
// Release: Prints 0 line(s)
using System;
using System.Runtime.CompilerServices;
using System.Numerics;
using System.Runtime.Intrinsics;
using System.Runtime.Intrinsics.X86;

public struct S0
{
    public int F0;
}

public struct S1
{
    public S0 F5;
}

public class Program
{
    public static S1 s_3;
    public static void Main()
    {
        var vr17 = Vector128.CreateScalar(2558356441U);
        var vr18 = Vector128.Create(0, 3113514718U, 0, 0);
        var vr19 = Sse2.ShiftRightLogical(vr17, vr18);
        if (0 >= Sse2.ConvertToUInt32(vr19))
        {
            S0 vr20 = s_3.F5;
            System.Console.WriteLine(vr20.F0);
        }
    }
}
```

This is with the fix in #105487

### PR DESCRIPTION
**Pull request by EgorBo**
Fixes https://github.com/dotnet/runtime/issues/105558

### CODE REVIEW

File: src/coreclr/jit/gentree.cpp [hunk 1]
```diff
@@ -30912,9 +30912,9 @@ GenTree* Compiler::gtFoldExprHWIntrinsic(GenTreeHWIntrinsic* tree)
 
                             int64_t shiftAmount = otherNode->AsVecCon()->GetElementIntegral(TYP_LONG, 0);
 
-                            if ((genTypeSize(simdBaseType) != 8) && (shiftAmount > INT_MAX))
+                            if ((uint64_t)shiftAmount >= ((uint64_t)genTypeSize(simdBaseType) * BITS_PER_BYTE))
```
Comment (tannergooding) [inline src/coreclr/jit/gentree.cpp:30915]: Does this one also need to be fixed in valuenum? It's also doing:
```
                        uint64_t shiftAmount = GetConstantSimd16(arg1VN).u64[0];

                        if (genTypeSize(baseType) != 8)
                        {
                            if (shiftAmount > INT_MAX)
                            {
                                // Ensure we don't lose track the the amount is an overshift
                                shiftAmount = -1;
                            }
                            arg1VN = VNForIntCon(static_cast<int32_t>(shiftAmount));
                        }
                        else
                        {
                            arg1VN = VNForLongCon(static_cast<int64_t>(shiftAmount));
                        }
                    }
```

Comment (EgorBo) [reply src/coreclr/jit/gentree.cpp]: Ah good find! addressed

**Comment by jakobbotsch** [comment]
Do other shifts/rotates need similar fixes? I see a lot of similar issues but with other intrinsics on the latest CI run (https://dev.azure.com/dnceng-public/public/_build/results?buildId=756781&view=ms.vss-build-web.run-extensions-tab)

**Comment by EgorBo** [comment]
> Do other shifts/rotates need similar fixes? I see a lot of similar issues but with other intrinsics on the latest CI run (https://dev.azure.com/dnceng-public/public/_build/results?buildId=756781&view=ms.vss-build-web.run-extensions-tab)
> 

This one is fixed by this PR so I suggest we do case-by-case, or feel free to post issues with multiple snippets

**Review by jakobbotsch — APPROVED** [review]
LGTM, but I'm not too familiar with this code, so might want to wait for Tanner as well

**Review by tannergooding — APPROVED** [review]

**Pull request merged**